File: MATLAB/metrics/test/testAUC.m
function testAUC()
%TESTAUC   Test cases for area under the ROC curve
%   evaluation metric
%
%   Author: Ben Hamner (ben@benhamner.com)

fprintf('Testing AUC ...');

actual = [1; 0; 1; 1];
posterior = [0.32; 0.52; 0.26; 0.86];
score = auc(actual, posterior);
assert(abs(1/3-score) < eps);

actual = [1; 0; 1; 0; 1];
posterior = [0.9 0.1 0.8 0.1 0.7]';
score = auc(actual, posterior);
assert(abs(1-score) < eps);

actual = [0 1 1 0]';
posterior = [0.2 0.1 0.3 0.4];
score = auc(actual, posterior);
assert(abs(1/4-score) < eps);

actual = [1 1 1 1 0 0 0 0 0 0 0]';
posterior = ones(size(actual));
score = auc(actual, posterior);
assert(abs(1/2-score) < eps);

fprintf('AUC tests passed\n');

File: R/inst/tests/test-auc.r
test.auc <- function()
{
    checkEqualsNumeric(auc(c(1,0,1,1), c(.32,.52,.26,.86)), 1/3)
    checkEqualsNumeric(auc(c(1,0,1,0,1),c(.9,.1,.8,.1,.7)), 1)
    checkEqualsNumeric(auc(c(0,1,1,0),c(.2,.1,.3,.4)), 1/4)
    checkEqualsNumeric(auc(c(1,1,1,1,0,0,0,0,0,0),0*(1:10)), 0.5)    
}
